tests/render/test_page_renderer.py

```diff
@@ -522,3 +539,64 @@ def test_snapshot_ValidationResultsPageRenderer_render_with_run_info_at_start(
     assert (
         rendered_validation_results == ValidationResultsPageRenderer_render_with_run_info_at_start
     )
+
+
+def test_asset_name_is_part_of_resource_info_index(mocker: MockerFixture):
+    """
+    DefaultSiteIndexBuilder.add_resource_info_to_index_links_dict is what supplies the
+    the resource meta-data to the index page.
+    This test checks that the asset_name is being provided when checkpoints are run.
+    """
+    import great_expectations as gx
+    from great_expectations.render.renderer.site_builder import DefaultSiteIndexBuilder
+
+    context = gx.get_context(mode="ephemeral")
+
+    add_resource_info_spy = mocker.spy(
+        DefaultSiteIndexBuilder,
+        "add_resource_info_to_index_links_dict",
+    )
```

Comment:
Can we test this another way? The public API of the default site index builder isn't super well known this feels like more of an implementation detail?